spec/models/stacks_media_token_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+describe StacksMediaToken do
+  test_start_time = Time.zone.now
+  let(:id) { 'ab012cd3456' }
+  let(:file_name) { 'def' }
+  let(:user_ip) { '192.168.1.100' }
+  subject { described_class.new(id, file_name, user_ip) }
+
+  describe '#create_from_encrypted_string' do
+    it 'should build an object with the right fields' do
+      encrypted_token_str = subject.to_encrypted_string
+      token_from_encrypted_str = StacksMediaToken.create_from_encrypted_string encrypted_token_str
+
+      expect(token_from_encrypted_str.id).to eq id
+      expect(token_from_encrypted_str.file_name).to eq file_name
+      expect(token_from_encrypted_str.user_ip_addr).to eq user_ip
+      expect(token_from_encrypted_str.timestamp).to be >= test_start_time
+      expect(token_from_encrypted_str.timestamp).to be <= Time.zone.now
+    end
+  end
+
+  describe '#token_still_valid_for?' do
+    it 'returns true for a valid token' do
+      expect(subject.token_still_valid_for?(id, file_name, user_ip)).to eq true
+    end
+
+    it 'returns false if the ids do not match' do
+      expect(subject.token_still_valid_for?('zy098xv7654', file_name, user_ip)).to eq false
+    end
+
+    it 'returns false if the file names do not match' do
+      expect(subject.token_still_valid_for?(id, 'fed', user_ip)).to eq false
+    end
+
+    it 'returns false if the IP addresses do not match' do
+      expect(subject.token_still_valid_for?(id, file_name, '192.168.1.101')).to eq false
+    end
+
```

Comment:
test for returns false if it's missing one of the required params?  If there is an invalid value for params?